probe-rs/src/architecture/arm/dp/mod.rs

```diff
@@ -484,6 +533,72 @@ impl Register for TARGETID {
     const NAME: &'static str = "TARGETID";
 }
 
+bitfield! {
+    /// BASEPTR0, Initiale system address for the first component in the system (see ADI v6.0 B2.2.2)
+    #[derive(Clone)]
+    pub struct BASEPTR0(u32);
+    impl Debug;
+    /// Pointer least significant bits (bits [31:12] of the full address).
+    pub u32, ptr, _: 31, 12;
+    /// True if `ptr` specifies a valid base address.
```

Comment:
This confuses me a bit: How would the API that uses this ever return an invalid base address?

Reply:
Probe-rs should stop if the device’s DP does not provide a valid base ptr for the ADIv6 root romtable as there’d be no way for probe-rs to know where to start.

Reply:
What I meant is: The chip would have to be broken for this to be wrong, no?

Reply:
I believe so, unless a vendor decides to provide this value via documentation instead of this register.